jenkins/ci.suse.de/pipelines/openstack-ardana.Jenkinsfile

```diff
@@ -33,8 +33,13 @@ pipeline {
           if (ardana_env == '') {
             error("Empty 'ardana_env' parameter value.")
           }
-          if (env.reserved_env && reserved_env != null) {
-            env.ardana_env = reserved_env
+          if (reserve_env == 'true') {
```

Comment:
Unfortunately, we cannot use the shared library here, because we need a git clone for that, and the resource needs to be reserved before we get to that point. Chicken-and-egg situation.

Reply:
But to use this Jenkinsfile there already must be a checkout, before the lock is taken, can't we use that?

Reply:
You're right, there is a checkout made on the "master" node (Jenkins server) to retrieve the Jenkinsfile. However, there are only two ways we can tell Jenkins that we also want access to shared libraries in the same checkout:
- either we run the pipeline on the "master" node, or
- we setup the shared library separately, as a [global shared](https://jenkins.io/doc/book/pipeline/shared-libraries/#global-shared-libraries) library, but this has serious security implications, because all global shared libraries are allowed to run outside the sandbox